### PR TITLE
Converting file name validations to run as a pre-commit hook

### PR DESCRIPTION
**Pull request by RosenbergYehuda**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-7819
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-11382

## Description
Converted the following validations, to run in pre-commit instead of validate.
- XD100
- PR100

### CODE REVIEW

File: demisto_sdk/scripts/validate_content_path.py [hunk 1]
```diff
@@ -180,6 +180,16 @@ class InvalidXSIAMReportFileName(InvalidPathException):
     message = "Name of XSIAM report files must start with the pack's name, e.g. `myPack_report1.json`"
 
 
+class InvalidXSIAMDashboardFileName(InvalidPathException):
+    message = "An XSIAM dashboard file must have a .json extension and its name must start with the pack's name, e.g., myPack_dashboard1.json."
```
Comment (JasBeilin) [inline demisto_sdk/scripts/validate_content_path.py:184]:
```suggestion
    message = "Only .json and .png file extension are supported for XSIAM dashboard. File must be named  <pack_name>_<dashboard_name>.json."
```

Comment (JasBeilin) [inline demisto_sdk/scripts/validate_content_path.py:184]:
```suggestion
    message = "Only .json and .png file extension are supported for XSIAM dashboard. File must be named  <pack_name>_<dashboard_name>.json."
```

File: demisto_sdk/scripts/validate_content_path.py [hunk 2]
```diff
@@ -180,6 +180,16 @@ class InvalidXSIAMReportFileName(InvalidPathException):
...
+
+
+class InvalidXSIAMParsingRuleFileName(InvalidPathException):
+    message = (
+        "An XSIAM Parsing Rule file name must be the same as the parent folder name."
```
Comment (JasBeilin) [inline demisto_sdk/scripts/validate_content_path.py:189]:
```suggestion
    "Only .yml and .xif file extension are supported for XSIAM Parsing Rule. File must be named as the parent folder name."
```

**Review by JasBeilin — COMMENTED** [review]
Looks good.

File: .changelog/4476.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Moved PR100 and XD100 validations to run as a pre-commit hook. These validations ensure the ParsingRule and XSIAM Dashboard filenames are valid.
```
Comment (dorschw) [inline .changelog/4476.yml:2]: mention the error codes for both, see other similar release notes and match their format.

**Review by dorschw — APPROVED** [review]

File: .changelog/4476.yml (hunk 3 above)
Comment (dorschw) [inline .changelog/4476.yml:2]:
```suggestion
- description: Moved PR100 and XD100 validations to run the `validate-content-path` **pre-commit** hook. These validations ensure the ParsingRule and XSIAM Dashboard filenames are valid.
```

**Review by JasBeilin — APPROVED** [review]

**Pull request merged**